com.unity.sg2/Editor/GraphUI/DataModel/SGGraphModel.cs

```diff
@@ -7,133 +7,36 @@
-        {
-            // Remove the parentheses
-            if (sQuaternion.StartsWith("(") && sQuaternion.EndsWith(")"))
-            {
-                sQuaternion = sQuaternion.Substring(1, sQuaternion.Length - 2);
-            }
-
-            // split the items
-            string[] sArray = sQuaternion.Split(',');
-
-            // store as a Vector3
-            Quaternion result = new Quaternion(
-                float.Parse(sArray[0]),
-                float.Parse(sArray[1]),
-                float.Parse(sArray[2]),
-                float.Parse(sArray[3]));
-
-            return result;
-        }
-
-        [SerializeField]
-        private SerializedMesh serializedMesh = new ();
-
-        [NonSerialized]
-        Quaternion m_Rotation = Quaternion.identity;
-
-        public Quaternion rotation
-        {
-            get => m_Rotation;
-            set
-            {
-                m_Rotation = value;
-                EditorPrefs.SetString(RotationUserPrefKey, rotation.ToString());
-            }
-        }
-
-
-        [NonSerialized]
-        float m_Scale = 1.0f;
-
-        public float scale
-        {
-            get => m_Scale;
-            set
-            {
-                m_Scale = value;
-                EditorPrefs.SetFloat(ScaleUserPrefKey, m_Scale);
-            }
-        }
-
-        public Mesh mesh
-        {
-            get => serializedMesh.mesh;
-            set
-            {
-                serializedMesh.mesh = value;
-                EditorPrefs.SetString(MeshUserPrefKey, EditorJsonUtility.ToJson(serializedMesh));
-            }
-        }
-    }
-
-    class ShaderGraphModel : GraphModel
+    class SGGraphModel : GraphModel
     {
         [SerializeField]
         private SerializableGraphHandler graphHandlerBox = new();
         [SerializeField]
         private SerializableTargetSettings targetSettingsBox = new();
         [SerializeField]
-        private MainPreviewData mainPreviewData;
+        private SGMainPreviewModel mainPreviewModel;
         [SerializeField]
         private bool isSubGraph = false;
 
         internal GraphHandler GraphHandler => graphHandlerBox.Graph;
         internal ShaderGraphRegistry RegistryInstance => ShaderGraphRegistry.Instance;
         internal List<JsonData<Target>> Targets => targetSettingsBox.Targets; // TODO: Store the active editing target in the box?
         internal Target ActiveTarget => Targets.FirstOrDefault();
-        internal MainPreviewData MainPreviewData => mainPreviewData;
+        internal SGMainPreviewModel MainPreviewData => mainPreviewModel;
         internal bool IsSubGraph => CanBeSubgraph();
         internal string BlackboardContextName => Registry.ResolveKey<PropertyContext>().Name;
         internal string DefaultContextName => Registry.ResolveKey<ShaderGraphContext>().Name;
 
         [NonSerialized]
-        Dictionary<RegistryKey, SGNodeUIData> m_NodeUIData;
+        Dictionary<RegistryKey, SGNodeUIData> m_NodeUIData = new ();
```

Comment:
made this change and hotfixed it into sg2/main so it wouldn't be a conflict

Reply:
It should be fine. If the change is the same, the merge works.